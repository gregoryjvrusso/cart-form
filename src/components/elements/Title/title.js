import React from "react";
import { array } from "prop-types";
import Circle from "../Circle";
import Flex from "../Flex";
import { TextTitle } from "./title.styles";
import { card } from "../../../assets/index";

const Title = ({ title }) => (
  <Flex p={2} justifyContent="center" alignItems="center">
    <Circle size="50" color="white">
      <img alt="card" src={card} />
    </Circle>
    <TextTitle as={"h1"}>
      {title[0]} <br />
      {title[1]}
    </TextTitle>
  </Flex>
);

Title.propTypes = {
  title: array.isRequired,
};

Title.displayName = "Title";


export default Title;
