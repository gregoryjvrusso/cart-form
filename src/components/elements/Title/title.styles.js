import styled from "styled-components";
import Text from "../Text";

import { theme } from "styled-tools";

export const TextTitle = styled(Text)`
  color: ${theme("colors.texts.white")};
  font-size: ${theme("fontSizes.4")}px;
  font-weight: ${theme("fontWeights.bold")};
  letter-spacing: 1px;
  margin: ${theme("spaces.2")}px;
`;
