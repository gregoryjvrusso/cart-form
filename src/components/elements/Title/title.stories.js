import React from "react";
import { storiesOf } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import theme from "../../../common/theme";

import Title from ".";
import Flex from "../Flex";

const dataMock = [
  "Adicione um novo",
  "cartão de crédito"
]
storiesOf("Elements/Title", module).add("default", () => (
  <ThemeProvider theme={theme}>
    <Flex bg="red">
      <Title title={dataMock} />
    </Flex>
  </ThemeProvider>
));
