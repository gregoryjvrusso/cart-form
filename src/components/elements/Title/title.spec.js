import React from "react";
import { render } from "@testing-library/react";
import Title from "../Title";

const dataMock = ["Adicione um novo", "cartão de crédito"];

describe("Title component", () => {
  it("deve conseguir renderizar [snapshot]", async () => {
    const { container } = render(<Title title={dataMock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("verificar se os valores do css estão corretos", () => {
    const { getByTestId } = render(<Title title={dataMock} />);
    expect(getByTestId("text-title").textContent.trim()).toContain(dataMock[0]);
  });
});
