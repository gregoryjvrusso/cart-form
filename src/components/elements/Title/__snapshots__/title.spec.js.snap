// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component deve conseguir renderizar [snapshot] 1`] = `
<div
  class="sc-AxjAm fdSDWf sc-AxirZ lbZIck"
  data-testid="flex"
>
  <div
    class="sc-AxjAm jndxDJ sc-AxirZ etMrKN sc-AxiKw yQzSy"
    color="white"
    data-testid="circle"
    size="50"
  >
    <img
      alt="card"
      src="card.svg"
    />
  </div>
  <h1
    class="sc-AxhUy bDxBeX"
    data-testid="text-title"
  >
    Adicione um novo
     
    <br />
    cartão de crédito
  </h1>
</div>
`;
